app/policies/rule_policy.rb

```diff
@@ -3,11 +3,13 @@
 # Policies for accessing Rules
 class RulePolicy < ApplicationPolicy
   def index?
-    record.profiles.pluck(:account_id).include? user.account_id
+    (Profile.canonical | record.profiles)
+      .pluck(:account_id).include? user.account_id
   end
 
   def show?
-    record.profiles.pluck(:account_id).include? user.account_id
+    (Profile.canonical | record.profiles)
+      .pluck(:account_id).include? user.account_id
```

Comment:
Pretty much a corner case, but if `account_id` is nil, this would still allow the user to see stuff. Thoughts about incorporating some change like this?

```ruby
Profile.where(id: (Profile.canonical | Rule.last.profiles)).distinct.pluck(:account_id)
``` 

In an example testing this, it produced a cleaner set:
```
["82658489-23cf-474e-827e-bb90c79d4cd9", "477d5248-f57b-489a-9c0a-c38f1c7b2ae5", "f6ca1f61-d03b-4f32-8118-c7c87766ec99"]
```
vs (I called `compact` on this because there were 30 instances of `nil` in the list and it was to long)
```
["82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "477d5248-f57b-489a-9c0a-c38f1c7b2ae5",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "477d5248-f57b-489a-9c0a-c38f1c7b2ae5",
 "f6ca1f61-d03b-4f32-8118-c7c87766ec99",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "82658489-23cf-474e-827e-bb90c79d4cd9",
 "477d5248-f57b-489a-9c0a-c38f1c7b2ae5"]
```